### PR TITLE
Support emptyDir as tieredstore's optional volume type for AlluxioRuntime

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
Support emptyDir as tieredstore's optional volume type

### Ⅱ. Does this pull request fix one issue?

#2069

Only for AlluxioRuntime now, will fixes it when all the runtime supports it.

### Ⅳ. Describe how to verify it
1. A naive AlluxioRuntime with modification on `tieredstore.levels[*].volumeType` can be used to verify it. 
```
apiVersion: data.fluid.io/v1alpha1
kind: AlluxioRuntime
metadata:
  name: demo-dataset
spec:
  replicas: 1
  tieredstore:
    levels:
      - mediumtype: SSD
        volumeType: emptyDir
        path: /mnt/ssd1,/mnt/ssd2
        #quota: 20Gi
        quotaList: 5Gi,10Gi
        high: "0.99"
        low: "0.99"
```

2. Check worker pod and fuse pod's `volumes` and `volumeMounts`

3. See something like this
```
  volumes:
  - emptyDir:
      medium: Memory
      sizeLimit: 10Gi
    name: mem
```

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: charts/alluxio/templates/_helpers.tpl [hunk 1]
```diff
@@ -292,6 +292,82 @@ resources:
   {{- end }}
 {{- end -}}
 
+{{- define "alluxio.fuse.tieredstoreVolumeMounts" -}}
```
Comment (cheyang) [inline charts/alluxio/templates/_helpers.tpl:295]: Please update chart version. Thanks.

File: pkg/ddc/alluxio/transform.go [hunk 2]
```diff
@@ -465,3 +460,27 @@ func (e *AlluxioEngine) transformPlacementMode(dataset *datav1alpha1.Dataset, va
 	}
 
 }
+
```
Comment (cheyang) [inline pkg/ddc/alluxio/transform.go:463]: Please add unit test. Thanks.

Comment (TrafalgarZZZ) [reply pkg/ddc/alluxio/transform.go:463]: Done

File: charts/alluxio/templates/_helpers.tpl (hunk 1 above)
Comment (TrafalgarZZZ) [reply charts/alluxio/templates/_helpers.tpl:295]: Done

File: api/v1alpha1/alluxioruntime_types.go [hunk 3]
```diff
@@ -157,6 +157,12 @@ type Level struct {
 	// +required
 	MediumType common.MediumType `json:"mediumtype"`
 
+	// VolumeType is the volume type of the tier. Should be one of the two types: `hostPath`, `emptyDir`.
+	// If not set, defaults to hostPath.
+	// +kubebuilder:default=hostPath
+	// +optional
+	VolumeType common.VolumeType `json:"volumeType"`
```
[the remark this answers is not in the document]
Comment (TrafalgarZZZ) [reply api/v1alpha1/alluxioruntime_types.go:165]: Done

**Review by cheyang — APPROVED** [review]

**Pull request merged**